File: mojo/tools/devtools.py
# Copyright 2015 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

import os
import sys


def add_lib_to_path():
  """ Adds the devtools pylib to path, allowing to use it in the internal
  /mojo/tools/ tooling. """
  sys.path.append(os.path.join(os.path.dirname(__file__),
                               os.pardir,
                               "devtools",
                               "common"))

File: .pub-cache/hosted/pub.dartlang.org/dwds-15.0.0/lib/src/servers/devtools.dart
// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import 'dart:io';

typedef DevtoolsLauncher = Future<DevTools> Function(String hostname);

/// A server for Dart Devtools.
class DevTools {
  final String hostname;
  final int port;
  final HttpServer _server;

  /// Null until [close] is called.
  ///
  /// All subsequent calls to [close] will return this future.
  Future<void>? _closed;

  DevTools(this.hostname, this.port, this._server);

  Future<void> close() => _closed ??= _server.close();
}
